aws/cluster.tf

```diff
@@ -14,15 +15,73 @@ locals {
   }
 }
 
+resource "aws_iam_policy" "redpanda" {
+  count  = var.tiered_storage_enabled ? 1 : 0
+  name   = local.deployment_id
+  path   = "/"
+  policy = jsonencode({
+    Version   = "2012-10-17"
+    Statement = [
+      {
+        "Effect": "Allow",
+        "Action": [
+          "s3:*",
+          "s3-object-lambda:*",
```

Comment:
This needs to be more limited to list only those needed permissions.

Reply:
Discussed this with the broader team. From that conversation:
> Having an policy that tightly limits redpanda to its own bucket is clearly the right thing to do, it's less obvious to me that restricting the verbs is helpful: it creates maintenance burden to keep those up to date as/when we change redpanda, and doesn't meaningfully change security if redpanda is the owner of the bucket.

So limiting this policy to the specific bucket (as shown in this file below) is the best approach, and avoids a maintenance burden as tiered storage capabilities are expanded.